src/backend/version-tracker/version-tracker.lambda.ts

```diff
@@ -0,0 +1,167 @@
+    metrics.putMetric(MetricName.TRACKED_PACKAGES_COUNT, versionMap.size, Unit.Count);
+    metrics.putMetric(MetricName.TRACKED_VERSIONS_COUNT, totalVersions, Unit.Count);
+  })();
+
+  // Upload the result to S3 and exit.
+  const result = await aws.s3().putObject({
+    Bucket: VERSION_TRACKER_BUCKET_NAME,
+    Key: VERSION_TRACKER_OBJECT_KEY,
+    Body: JSON.stringify(versionJson),
+    ContentType: 'application/json',
+    CacheControl: 'public, max-age=60, must-revalidate, proxy-revalidate', // Expire from cache after 1 minute
+    Metadata: {
+      'Lambda-Log-Group': context.logGroupName,
+      'Lambda-Log-Stream': context.logStreamName,
+      'Lambda-Run-Id': context.awsRequestId,
+      'Package-Count': `${versionMap.size}`,
+      'Version-Count': `${totalVersions}`,
+    },
+  }).promise();
+
+  return result;
+}
+
+/**
+ * List all prefixes in a bucket given a base prefix.
+ *
+ * For example, if given "data/" it will return
+ * [
+ *   "data/@aws-amplify/",
+ *   "data/@aws-c2a/",
+ *   "data/@aws-cdk/",
+ *   ...
+ * ]
+ */
+async function listPrefixes(bucket: string, prefix: string): Promise<string[]> {
+  const prefixes = new Array<string>();
+
+  let continuationToken;
+  do {
+    const listRequest: AWS.S3.ListObjectsV2Request = {
+      Bucket: bucket,
+      Prefix: prefix,
+      Delimiter: '/',
+      ContinuationToken: continuationToken,
+    };
+    const listResponse = await aws.s3().listObjectsV2(listRequest).promise();
+    continuationToken = listResponse.NextContinuationToken;
+
+    for (const { Prefix: commonPrefix } of listResponse.CommonPrefixes ?? []) {
+      if (!commonPrefix) { continue; }
+      prefixes.push(commonPrefix);
+    }
+
+  } while (continuationToken);
+
+  return prefixes;
+}
+
+/**
+ * List all package prefixes in a bucket given a base prefix.
+ *
+ * For example, if given "data/" it will return
+ * [
+ *   "data/"
+ * ]
+ */
+async function listPackagePrefixes(bucket: string, prefix: string): Promise<string[]> {
+  const packagePrefixes = new Array<string>();
+
+  // gather a list of all package scopes and unscoped packages
+  const initialPrefixes = await listPrefixes(bucket, prefix);
+  const scopedPrefixes = initialPrefixes.filter((p) => (p?.startsWith('data/@') && prefix === 'data/'));
+  const unscopedPrefixes = initialPrefixes.filter((p) => !(p?.startsWith('data/@') && prefix === 'data/'));
+
+  // scoped packages need to be collected separately, so we
+  // group the requests into batches, run them in parallel, and
+  // flatten the results to an output list
+  const batches = groupIntoBatches(scopedPrefixes, BATCH_SIZE);
+  for (const batch of batches) {
+    const promises: Promise<string[]>[] = batch.map(async (scopedPrefix) => listPrefixes(bucket, scopedPrefix));
+    const results = await Promise.all(promises);
+    packagePrefixes.push(...results.flat());
+  }
+
+  for (const packagePrefix of unscopedPrefixes) {
+    packagePrefixes.push(packagePrefix);
+  }
```

Comment:
```suggestion
  packagePrefixes.push(...unscopedPrefixes);
```

There's also `Array.concat`...